akka-http-core/src/main/scala/akka/http/scaladsl/model/ws/UpgradeToWebSocket.scala

```diff
@@ -88,6 +93,19 @@ trait UpgradeToWebSocket extends jm.ws.UpgradeToWebSocket {
     subprotocol: String): HttpResponse =
     handleMessages(createScalaFlow(inSink, outSource), subprotocol = Some(subprotocol))
 
-  private[this] def createScalaFlow(inSink: Graph[SinkShape[jm.ws.Message], _ <: Any], outSource: Graph[SourceShape[jm.ws.Message], _ <: Any]): Graph[FlowShape[Message, Message], NotUsed] =
-    JavaMapping.toScala(scaladsl.Flow.fromSinkAndSourceMat(inSink, outSource)(scaladsl.Keep.none): Graph[FlowShape[jm.ws.Message, jm.ws.Message], NotUsed])
+  override def handleMessagesWithSink(sink: Graph[SinkShape[jm.ws.Message], _ <: Any]): HttpResponse =
+    handleMessages(createScalaCoupledFlow(sink, Source.maybe))
+
+  override def handleMessagesWithSource(source: Graph[SourceShape[jm.ws.Message], _ <: Any])(implicit mat: Materializer): HttpResponse =
+    handleMessages(createScalaCoupledFlow(akka.http.javadsl.model.ws.WebSocket.ignoreSink(mat), scaladsl.Source.fromGraph(source)))
+
+  private[this] def createScalaFlow(inSink: Graph[SinkShape[jm.ws.Message], _ <: Any], outSource: Graph[SourceShape[jm.ws.Message], _ <: Any]): Graph[FlowShape[Message, Message], NotUsed] = {
+    val graph: Graph[FlowShape[jm.ws.Message, jm.ws.Message], NotUsed] = scaladsl.Flow.fromSinkAndSourceMat(inSink, outSource)(scaladsl.Keep.none)
+    JavaMapping.toScala(graph)
+  }
+
+  private[this] def createScalaCoupledFlow(inSink: Graph[SinkShape[jm.ws.Message], _ <: Any], outSource: Graph[SourceShape[jm.ws.Message], _ <: Any]): Graph[FlowShape[Message, Message], NotUsed] = {
+    val graph: Graph[FlowShape[jm.ws.Message, jm.ws.Message], NotUsed] = scaladsl.CoupledTerminationFlow.fromSinkAndSource(Sink.fromGraph(inSink), Source.fromGraph(outSource)).mapMaterializedValue(_ ⇒ NotUsed)
+    JavaMapping.toScala(graph)
+  }
```

Comment:
hah thanks, I did not notice the email about this pr after the morning flood of emails :( Managed to solve it though. Thanks and sorry I missed this